src/common/ui-components/Search/Search.tsx

```diff
@@ -86,13 +86,18 @@ const Search: React.FC<SearchProps> = ({
   const { t } = useTranslation();
   const [searchResults, setSearchResults] = useState<SearchData[]>([]);
   const [currentSearch, setCurrentSearch] = useState<string>('');
+  const [showInput, setShowInput] = useState(false);
```

Comment:
Not sure why state `showInput` is needed as there's already a prop `isMenuOpen` which accomplishes the same thing? Also `isMenuOpen` should probably be renamed to `showInput`.

Reply:
I think keeping them separate is alright for now, it also could be used for the animation effects. It logically just made more sense to me.

Reply:
The point was they are not separate. You can just replace every instance of `showInput` with `isMenuOpen` and it would logically be the same, without the code needed to sync the two?

Reply:
I don`t think we should toggle the menu state through the search component, though, that was my side of it. I ll do the way you ask as a compromise, but I personally find this comment to be nit-picking.

Reply:
After considering your approach, which I partially agree with, of course, I believe it brings slightly more complexity to replace input state with isMenuOpen considering they are separate animations. I understand you point, and I am addressing it in this way: I believe we should leave it the way it is for now.